### ISSUE TITLE
Password validation on Reauthentication Form

### ISSUE THREAD

**Issue by LCMJ21** [pennersr/django-allauth#3493]
I recently added mfa support for my project and i was forced to add reauthentication but i encountered a problem with the reauthentication form:

Password validation is made by using `check_password` method, but in my project's case, my login is made by calling an external API, so this does not work on integrations with external apps.

```
def clean_password(self):
        password = self.cleaned_data.get("password")
        if not self.user.check_password(password):
            raise forms.ValidationError(
                get_adapter().error_messages["incorrect_password"]
            )
        return password
```

I can rewrite the form but what if, instead of `check_password` method, the form used the adapter's `authenticate` method ?

**Comment by pennersr** [comment]
It would definitely make sense to look into using the adapter's `authenticate()` somehow, as that one also contains rate limiting which the current reauthentication does not support. Though:

- `authenticate()` refers to `LOGIN_ATTEMPTS_LIMIT` and `self.error_messages["too_many_login_attempts"]`. Yet, we're not logging in. TBD if this is an issue.
- For reauthentication the user is already known. For `authenticate()` the user is not known, and would be indirectly have to be looked up by email or username.

Given the above, perhaps a `reauthenticate()` method would be a better choice.

Do you call the external API from the adapter `authenticate()` or from a custom authentication backend?

**Comment by LCMJ21** [comment]
First, sorry for the delay in the response.

Yeah, I agree with you, a `reauthenticate method` would be a better choice. I have a custom backend class for the API that is declared in the Django `AUTHENTICATION_BACKENDS` variable, so I suggested the authenticate method because it uses all backends defined there. So yes, a `reauthenticate` method that would use all backends defined in `AUTHENTICATION_BACKENDS` I think would be the correct solution

**Comment by pennersr** [comment]
Closed via df41f286

**Comment by pennersr** [comment]
https://codeberg.org/allauth/django-allauth/issues/3493
